### DISCUSSION TITLE
X-Touch events not registered/populating MIDI2LR at all

### DISCUSSION THREAD

**Discussion by godotozozozo** [rsjaffe/MIDI2LR#559] [Problems]
Hello--

I've previously used MIDI2LR with a Behringer X-Touch mini and Lightroom CC without any problems for a while, but the most recent re-install on same machine has been a complete no-go.

First off, the setup gives me a warning about ports 58763-58764, even though nothing is using the ports during the setup, or during using Lightroom--
<img width="400" alt="2021-11-22 08_40_32-Warning" src="https://user-images.githubusercontent.com/94848303/142876731-23c4e097-59b9-4538-87ff-936101697d91.png">
<img width="600" alt="midi2lr ports" src="https://user-images.githubusercontent.com/94848303/142876808-21b9b23d-386b-4468-85dc-8f1c7009f8ed.png">

Then, the plug-in loads fine but shows nothing in the interface. Moving the knobs, nothing pops up in the list. I tried loading my old saved presets, same thing.

<img width="500" alt="midi2lr interface" src="https://user-images.githubusercontent.com/94848303/142876824-fe4b0c82-a1aa-4a3b-af9e-560ddaacd1d0.png">

I tried opening the appdata/roaming folder and found a "DisabledControllers.xml" file which lists the X-Touch as well as the other midi devices, and seems to have them all disabled? 

<img width="1200" alt="midi2lr disabled controllers" src="https://user-images.githubusercontent.com/94848303/142876853-bbe66a9a-a40c-4b1e-92cc-bcd35030ab48.png">

Deleting anything from that xml file, or changing values to zero made no difference.

Tried the most recent version of the plugin, 5.1.1.0, as well as downgrading to the previous one, no difference.

Any thoughts on what to try?

**Comment by rsjaffe** [comment]
1. You can delete _DisabledControllers.xml_. That way it will reload a fresh version. Unless you had changed _active_ to 0, it shouldn't have had any contribution to the problem, but deleting it completely is safest.
2. Is your controller plugged in and turned on (I've done this!). Check USB connection.
3. Use a fresh USB cable--could have a bad cable.
4. Click _Rescan MIDI Devices_ on the app and see if it's then fixed.
5. Failing that, post a copy of MIDI2LR.log for me to see.

**Comment by godotozozozo** [comment]
Thanks for the quick response!

I tried these out step by step -- step one seemed to work! and then not -- and wrote out a really detailed report before concluding that there's ultimately something erratic with the X-Touch hardware itself (that possibly messes up the MIDI channel it's on...?). The cable is fine (tried two, with different devices as well), and MIDI2LR is fine (other midi devices register; I have currently remapped a Komplete Kontrol 32 to use for it). 

X-Touch, when it shows up, is on channel 11 -- and it works until something throws it off -- still not sure what -- and then it doesn't register in the plugin or anywhere else. Unplugging it a few times gets it to work again, but I can't figure out what the culprit is. The actual USB port where cable plugs in? Some other random thing? In any case, I'll conclude it's a hardware fault and leave it at that.

thanks again for the support though, and while that particular controller's out, I'll make some new decals for the other one for the time being, and am glad to continue using the plugin!

t.

**Comment by prohidium** [comment]
Not sure if this is your situation but I have a BCR2000 that will not work properly when connected to a USB hub. It works fine directly on a USB2 port on the PC. The USB Hub is USB3 as well so that may be a factor. Maybe try plugging into a USB2 only port directly on the computer if you are plugged into a USB3 port.
